### PR TITLE
actions issue

### LINKED ISSUE #72661: Tests should be disabled in forks when they are expected to fail
### 🐛 Describe the bug

Currently many tests run in forks but they are expected to fail due to lack of credentials in forks.

This is a bit annoying to contributors:

A failed job example: https://github.com/ppwwyyxx/pytorch/runs/5137367784?check_suite_focus=true

I think github action should be configured to not run these in forks.

### Versions

latest as of 02/09

cc @seemethere @malfet @pytorch/pytorch-dev-infra

### PR DESCRIPTION
**Pull request by arunavabasucom**
Fixes #72661
Hey @janeyx99 please review it 😃

### CODE REVIEW

**Comment by janeyx99** [comment]
You have the right idea! Note here https://github.com/pytorch/pytorch/blob/master/.github/workflows/generated-ios-12-5-1-arm64-full-jit.yml#L3 that the file you edited is generated, and it would be more accurate to edit the generate_ci_worflows.py script instead. This is also why the CI lint is failing, as the generation script is the source of truth.

File: .github/workflows/generated-ios-12-5-1-arm64-full-jit.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
   build:
     # NOTE: These builds will not run successfully without running on `pytorch/pytorch` due to the limitations
     #       of accessing secrets from forked pull requests and IOS' dependency on secrets for their build/test
-    if: ${{ github.event_name == 'push' || github.event.pull_request.head.repo.full_name == github.repository }}
+    if: ${{ github.event_name == 'push' || github.event.pull_request.head.repo.full_name == github.repository || github.repository_owner == 'pytorch' }}
```
Comment (janeyx99) [inline .github/workflows/generated-ios-12-5-1-arm64-full-jit.yml:31]:
```suggestion
    if: ${{ github.event_name == 'push' || (github.event.pull_request.head.repo.full_name == github.repository && github.repository_owner == 'pytorch') }}
```
Should be an and, no?

**Review by janeyx99 — CHANGES_REQUESTED** [review]

**Comment by janeyx99** [comment]
Please don't hesitate to re-request a review once the review has been addressed

File: .github/workflows/generated-ios-12-5-1-arm64-full-jit.yml (hunk 1 above)
Comment (arunavabasu-03) [reply .github/workflows/generated-ios-12-5-1-arm64-full-jit.yml:31]: yeah that's my fault , i commit your suggeation

**Pull request closed without merging**